client/landing/jetpack-cloud/components/scan-threats/index.tsx

```diff
@@ -44,27 +45,52 @@ const ScanThreats = ( { site, threats }: Props ) => {
 	const dispatch = useDispatch();
 
 	const openFixAllThreatsDialog = React.useCallback( () => {
+		dispatch(
+			recordTracksEvent( `calypso_scan_all_threats_dialog_open`, {
+				site_id: site.ID,
+			} )
+		);
 		setShowFixAllThreatsDialog( true );
-	}, [] );
+	}, [ dispatch, site ] );
 
-	const openDialog = React.useCallback( ( action: ThreatAction, threat: Threat ) => {
-		setSelectedThreat( threat );
-		setActionToPerform( action );
-		setShowThreatDialog( true );
-	}, [] );
+	const openDialog = React.useCallback(
+		( action: ThreatAction, threat: Threat ) => {
+			dispatch(
+				recordTracksEvent( `calypso_scan_${ action }_threat_dialog_open`, {
+					site_id: site.ID,
+					threat_id: threat.id,
```

Comment:
Same as below. I don't think we need the `threat.id` here.

Reply:
Should we also record the close event?

Reply:
At first, I added those events as well and then I removed them because I don't have a sense yet for how detailed our tracking system should be.